tensorflow/lite/g3doc/inference_with_metadata/task_library/audio_classifier.md

```diff
@@ -101,6 +101,134 @@ See the
 [source code and javadoc](https://github.com/tensorflow/tflite-support/blob/master/tensorflow_lite_support/java/src/java/org/tensorflow/lite/task/audio/classifier/AudioClassifier.java)
 for more options to configure `AudioClassifier`.
 
+## Run inference in iOS
+
+### Step 1: Install the dependencies
+
+The Task Library supports installation using CocoaPods. Make sure that CocoaPods
+is installed on your system. Please see the
+[CocoaPods installation guide](https://guides.cocoapods.org/using/getting-started.html#getting-started)
+for instructions.
+
+Please see the
+[CocoaPods guide](https://guides.cocoapods.org/using/using-cocoapods.html) for
+details on adding pods to an Xcode project.
+
+Add the `TensorFlowLiteTaskAudio` pod in the Podfile.
+
+```
+target 'MyAppWithTaskAPI' do
+  use_frameworks!
+  pod 'TensorFlowLiteTaskAudio'
+end
+```
+
+Make sure that the `.tflite` model you will be using for inference is present in
+your app bundle.
+
+### Step 2: Using the model
+
+#### Swift
+
+```swift
+// Imports
+import TensorFlowLiteTaskAudio
+import AVFoundation
+
+// Initialization
+guard let modelPath = Bundle.main.path(forResource: "sound_classification",
+                                            ofType: "tflite") else { return }
+
+let options = AudioClassifierOptions(modelPath: modelPath)
+
+// Configure any additional options:
+// options.classificationOptions.maxResults = 3
+
+let classifier = try AudioClassifier.classifier(options: options)
+
+// Create Audio Tensor to hold the input audio samples which are to be classified.
+// Created Audio Tensor has audio format matching the requirements of the audio classifier.
+// For more details, please see:
+// https://github.com/tensorflow/tflite-support/blob/master/tensorflow_lite_support/ios/task/audio/core/audio_tensor/sources/TFLAudioTensor.h
+let audioTensor = classifier.createInputAudioTensor()
+
+// Create Audio Record to record the incoming audio samples from the on-device microphone.
+// Created Audio Record has audio format matching the requirements of the audio classifier.
+// For more details, please see:
+https://github.com/tensorflow/tflite-support/blob/master/tensorflow_lite_support/ios/task/audio/core/audio_record/sources/TFLAudioRecord.h
+let audioRecord = try classifier.createAudioRecord()
+
+// Request record permissions from AVAudioSession before invoking audioRecord.startRecording().
+AVAudioSession.sharedInstance().requestRecordPermission { granted in
+    if granted {
+        DispatchQueue.main.async {
```

Comment:
@khanhlvg `AVAudioSession.sharedInstance().requestRecordPermission` on the first run of the app, after it waits for the user's input for permissions, somehow does not execute the completion block. In the subsequent runs it is fine. This is unrelated to our implementation. Hence had to dispatch to a different queue or main in the example app. I have retained this here, because developers may not figure it out easily.
But this can be removed since it's not the task library's issue. It's an `AVFoundation` bug, I guess. LMK what needs to be done.